app/controllers/api/admins_controller.rb

```
@@ -2,50 +2,106 @@
 
 class Api::AdminsController < ApplicationController
   before_action :authenticate_admin
-  skip_before_action :authenticate_admin, only: [ :login, :refresh_token ]
+  skip_before_action :authenticate_admin, only: [:login, :refresh_token]
+
+  def transactions
+      begin
+      transactions = DriverBalanceTransaction.all
+      render_json(ActiveModelSerializers::SerializableResource.new(
+          transactions.page(pagination[:page]).per(pagination[:per_page]),
+          each_serializer: TransactionSerializer),
+                  status: :ok,
+                  meta: { total: transactions.length },
+                  message: "Transactions fetched successfully"
+      )
+    rescue Exception => e
+      raise Errors::ApplicationError, e.message
+    end
+  end
+
+  def update_transaction
+
+  end
 
   def review_kyc
-    command = AdminCommands::ReviewKycCommand.call(@current_admin, params[:driver_id], params[:status], params[:review])
+    command = AdminCommands::ReviewKycCommand.call(@current_admin, params[:driver_id], params[:status], params[:kyc_review])
     if command.success?
-      render json: { data: command.result }, status: :ok
+      render json: { message: "Review successful", data: nil }, status: :ok
     else
       raise Errors::Invalid, "Review failed"
     end
   end
 
   def get_drivers
-    render json: { data: Driver.all }, status: :ok
+    begin
+      drivers = Driver.all
+      render_json(ActiveModelSerializers::SerializableResource.new(
+          drivers.page(pagination[:page]).per(pagination[:per_page]),
+          each_serializer: DriverSerializer, show_more_info: "detail"),
+                  status: :ok,
+                  meta: { total: drivers.length },
+                  message: "Drivers fetched successfully"
+      )
+    rescue Exception => e
+      raise Errors::ApplicationError, e.message
```

let it crash and through exception, handle exception notification later different way